lms/envs/common.py

```diff
@@ -587,7 +587,7 @@
 # Note: these intentionally greedily grab all chars up to the next slash including any pluses
 # DHM: I really wanted to ensure the separators were the same (+ or /) but all patts I tried had
 # too many inadvertent side effects :-(
-COURSE_KEY_PATTERN = r'(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/]+)'
+COURSE_KEY_PATTERN = r'(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/][^?/]*)'
```

Comment:
Oh, I see. I had the diff in side-by-side mode, and wasn't seeing exactly the change you made.

This seems a bit odd. Won't it match a string like `org/course/?`, and treat the `?` as part of the course key? It would be worth verifying that `org/course/?foo=bar` doesn't match either.